src/ansys/fluent/core/solver/flobject.py

```diff
@@ -1010,8 +1058,14 @@ def get_completer_info(self, prefix="", excluded=None) -> List[List[str]]:
 class Command(Action):
     """Command object."""
 
-    def __call__(self, **kwds):
-        """Call a query with the specified keyword arguments."""
+    def __call__(self, purpose: Optional[str] = None, **kwds):
+        """Call a command with the specified keyword arguments."""
+        for arg, value in kwds.items():
+            argument = getattr(self, arg)
+            if (
+                hasattr(argument, "is_input") and argument.is_input()
+            ) or purpose == "input":
+                argument.before_execute(value)
```

Comment:
This is different to the design I shared with you.

Reply:
If we don't add `if (hasattr(argument, "is_input") and argument.is_input())` check then we get attribute error for arguments other than `file_name` or `file_name_list`. 

Current Fluent changes are not in Ansys Lab therefore we have added `or purpose == "input"` condition to test it.